### PR TITLE
Use more idiomatic or_else construction

### LINKED ISSUE #15812: Use Option::or_else instead of transforming Some => Some
We have code inside matches in ThreadSafeLayoutNodeChildrenIterator that looks like `Some(x) => Some(x)`. We can replace the `match foo` with `foo.or_else(|| ...)` where the ... is the code for the None branch. This is more idiomatic Rust code.

Code: `components/script/layout_wrapper.rs`

### PR DESCRIPTION
**Pull request by tvladyslav**
or_else in more ideomatic rust construction than
match, where Some(x) => Some(x)
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #15812 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15815)

### CODE REVIEW

File: components/script/layout_wrapper.rs [hunk 1]
```diff
@@ -953,19 +953,12 @@ impl<ConcreteNode> Iterator for ThreadSafeLayoutNodeChildrenIterator<ConcreteNod
...
+                            self.parent_node.get_details_summary_pseudo()
+                                .or_else(|| unsafe { self.parent_node.dangerous_first_child() })
+                                .or_else(|| self.parent_node.get_after_pseudo())
                         },
                         PseudoElementType::Normal => {
-                            match unsafe { node.dangerous_next_sibling() } {
-                                Some(next) => Some(next),
-                                None => self.parent_node.get_after_pseudo(),
-                            }
+                            (unsafe { node.dangerous_next_sibling() }).or_else(|| self.parent_node.get_after_pseudo())
```
Comment (KiChjang) [inline components/script/layout_wrapper.rs]: nit: The parenthesis around `unsafe` should be unnecessary.

**Review by KiChjang — CHANGES_REQUESTED** [review]
Great work! The code styling looks correct, but if you just address one minor nit, we can land this!

**Comment by KiChjang** [comment]
Also, there's a typo in the commit message, `idiomatic` instead of `ideomatic`.

**Comment by crypto-universe** [comment]
@KiChjang fixed. Check PR now.

**Comment by KiChjang** [comment]
@bors-servo r+

Thanks!

**Review by KiChjang — APPROVED** [review]

**Comment by crypto-universe** [comment]
Thank you!

**Pull request merged**